SparkPost/2web2ui
# FE-1204 - UX Feedback for Domain Detail Pages Part - 2

### What Changed
 - Disable the `Authenticate for Bounce` and `Authenticate for SPF` buttons until they select the checkbox
 - Disable the `Update Tracking Domain` button unless the user changes the select value
 - Add the checkbox confirmation to the verify-tracking page: 
![image](https://user-images.githubusercontent.com/4204806/96503941-9f3d2900-1219-11eb-882a-2c84b92eecd1.png)
(and disable the button till they click the checkbox)
 - Add the confirmation alert when toggling domain subaccount sharing. 
![image](https://user-images.githubusercontent.com/4204806/96505787-3efbb680-121c-11eb-8c28-21533788eab0.png)
![image](https://user-images.githubusercontent.com/4204806/96505809-458a2e00-121c-11eb-89aa-3f29ffd2a1cd.png)


### How To Test
(The hardest part of testing this is having domains that render the sections in question - locally I forced the components to show up when I needed)
 - Go to the Domain Details page `http://localhost:3100/domains/details/{domain_name}`
 - Go to the verify tracking details page: `http://localhost:3100/domains/details/{domain_name}/verify-tracking`

Verify each button changed in this PR is disabled until the checkbox is selected or the select dropdown value changes. 

List of files that have user facing impact. The rest of the files are jest/unit testing updates.
```
src/pages/domains/components/TrackingDnsSection.js
src/pages/domains/components/SetupForSending.js
src/pages/domains/components/SetupBounceDomainSection.js
src/pages/domains/components/SendingAndBounceDomainSection.js
src/pages/domains/components/LinkTrackingDomainSection.js
src/pages/domains/components/DomainStatusSection.js
```


### To Do
- [ ] Review Feedback
